src/lib.rs

```diff
@@ -572,17 +613,17 @@ pub fn init(options: &InitOptions) -> Result<(ZWaveManager, mpsc::Receiver<ZWave
     try!(zwave_manager.add_watcher());
 
     let device = match options.device {
-        Some(ref device) => device as &str,
+        Some(ref device) => device.to_owned(),
```

Comment:
maybe simpler to understand: `Some(device) => device.clone()`. I don't know if this builds :)